typeid/typeid-go/typeid.go

```diff
@@ -0,0 +1,62 @@
+package typeid
+
+import (
+	"github.com/gofrs/uuid"
+	"go.jetpack.io/typeid/base32"
+)
+
+type TypeID struct {
+	prefix string
+	suffix string
+}
+
+var Nil = TypeID{
+	prefix: "",
+	suffix: "00000000000000000000000000",
+}
+
+func New(prefix string) (TypeID, error) {
+	uid, err := uuid.NewV7()
+	if err != nil {
+		return Nil, err
+	}
+	tid := TypeID{
+		prefix: prefix,
+		suffix: base32.Encode(uid),
+	}
+	return tid, nil
+}
+
+func (tid TypeID) Type() string {
+	return tid.prefix
+}
+
+func (tid TypeID) Suffix() string {
+	return tid.suffix
+}
+
+func (tid TypeID) String() string {
+	if tid.prefix == "" {
+		return tid.suffix
+	}
+	return tid.prefix + "_" + tid.Suffix()
+}
+
+func (tid TypeID) UUIDBytes() []byte {
+	b, err := base32.Decode(tid.suffix)
+	if err != nil {
+		panic(err)
```

Comment:
Hehe, why yikes?

This error will never happen since the constructors will be the ones throwing an error, and guaranteeing once and object is constructed the data is valid and can be decoded. I see it as the way to make the compiler realize the invariant so that the return type is accurate.

(that said, if we do get a panic, I'll eat my words and change the signature, lol)